tmt/steps/execute/internal.py

```diff
@@ -259,6 +262,11 @@ def go(self):
                 if self._handle_reboot(test, guest):
                     continue
                 self._results.append(self.check(test))
+                if exit_first and self._results[-1].result == 'fail':
```

Comment:
How about 'warn' and 'error'? Per docs all three results should be treated as fail (by automation). Also this option is more about 'first test that doesn't pass' than just an explicit fail (IMO).

 Also detached checks for non-zero exit_code which hits timeout (== error)  so this won't be consistent.

Reply:
Good catch. According to `Result._results`, there can also be `info` (which I have never encountered yet but I guess that can occur). Lets change this condition to `self._results[-1].result not in ('pass', 'info')` then.